src/qmflows/parsers/cp2KParser.py

```diff
@@ -261,34 +275,49 @@ def remove_trailing(xs):
 # Parsing From File
 
 #: A tuple with 2 elements; output of :func:`read_mos_data_input`.
-Tuple2 = Tuple[Optional_[str], Optional_[List[int]]]
+Tuple2 = Tuple[Optional_[int], Optional_[Tuple[int, int]]]
 
 
 def read_mos_data_input(path_input: PathLike) -> Tuple2:
     """Try to read the added_mos parameter and the range of printed MOs."""
     l1 = try_search_pattern("ADDED_MOS", path_input)
     l2 = try_search_pattern("MO_INDEX_RANGE", path_input)
 
-    added_mos = l1.split()[-1] if l1 is not None else None
-    range_mos = list(map(int, l2.split()[1:])) if l2 is not None else None
+    added_mos = int(l1.split()[-1]) if l1 is not None else None
+    range_mos = tuple(map(int, l2.split()[1:])) if l2 is not None else None
 
     return added_mos, range_mos
 
 
 def read_cp2k_number_of_orbitals(file_name: PathLike) -> MO_metadata:
     """Look for the line ' Number of molecular orbitals:'."""
-    def fun_split(string: str) -> str:
-        return string.split()[-1]
+    def fun_split(string: Optional_[str]) -> int:
+        if string is not None:
+            return int(string.split()[-1])
+        else:
+            return 0
 
     properties = ["Number of occupied orbitals", "Number of molecular orbitals",
                   "Number of orbital functions"]
 
-    try:
-        xs = [fun_split(try_search_pattern(x, file_name)) for x in properties]  # type: ignore
-    except AttributeError as ex:  # try_search_pattern() can return None
-        raise RuntimeError(f"Failed to identify orbitals in {file_name!r}") from ex
+    meta = MO_metadata(*[fun_split(try_search_pattern(x, file_name)) for x in properties])
+
+    if any(getattr(meta, att) == 0 for att in ("nOrbitals", "nOccupied", "nOrbFuns")):
```

Comment:
```suggestion
    if not all(meta[:-1]):
```
All elements are integers anyway, so `bool(i)` should be equivalent to `i == 0`.

Reply:
Right, thanks for the suggestion!